### PR TITLE
fix: Theming system, major UI bugs, redesign minimalist RC-like UI.

### LINKED ISSUE #539: [BUG]: Sidebar Overlapping with Chat Body
### Description:

In the current implementation, when opening the sidebar, it overlaps with the chat body, which is not good behavior. Instead, the chat body should adjust its width accordingly when the sidebar is open.

### Steps to reproduce:

1. Set up Embedded Chat.
2. Click on any option that opens the sidebar, such as Thread menu, Mention menu, Channel info menu, etc.
3. Notice that the sidebar overlaps with the chat body.

### Expected behavior:

The chat body width should adjust accordingly whenever the sidebar opens up.

### Actual behavior:
image

The sidebar overlaps with the chat body, giving the user less space to send messages or perform tasks.

### LINKED ISSUE #577: [Feat]: Addition of multiple themes to the project
As an EmbeddedChat developer,

**I need to**:
Design some good-looking themes using the existing theming foundation

**So that**:
EmbeddedChat admins have the option to choose designs that suit their website well upon integration

**Acceptance Criteria**

- [ ] Design multiple themes

### PR DESCRIPTION
**Pull request by Spiral-Memory**
# Brief Title
Fixing Theming System; Redesigning Minimalist RC-like UI

## Acceptance Criteria Fulfillment

- [x] Shifted to ShadCN naming system
- [x] Fixed the existing theming system
- [x] Fixed various component stylings
- [x] Limited modal size to EC component using React Portals
- [x] Removed Dropbox due to numerous bugs (will add later if required)
- [x] Fixed ImageGallery till EC component
- [x] Fixed UI inconsistencies and bugs
- [x] Fixed Sidebar overlap issue
- [x] Redesigned the UI of the first variant 'RocketChat - Minimalist'
- [x] Fixed Popup
- [x] Removed unnecessary dependencies
- [x] Fixed member list and command list popup
- [x] Fixed swiper bugs and stylings 
- [x] Redesigned and matched colors!

Fixes #577
Fixes #539

## Video/Screenshots

**Before:** (with no theming system, no dark mode, no multiple themes, lots of UI bugs, and inconsistent stylings)

https://github.com/RocketChat/EmbeddedChat/assets/78961432/3cccc9fc-2030-4545-a127-c865f7d0fcd5

**After:** (after all the mentioned fixes)

https://github.com/RocketChat/EmbeddedChat/assets/78961432/d9fea331-fa32-44df-8322-36c9fb2baa6c

### CODE REVIEW

**Comment by Spiral-Memory** [comment]
Hey @sidmohanty11 

This PR is now ready for review! It mainly focuses on resolving all major UI bugs in the app and making the app consistent with theming colors, z-index, shadows, etc., as defined in the theme files. Previously, these were applied randomly, resulting in a very messy project with inconsistent colors throughout.

I've done my best to ensure everything matches across multiple themes (refer to the video). Dark mode is also supported in all themes.

There might still be some remaining bugs, which I will fix in subsequent PRs. Additionally, there is quite a bit of duplicated code at the moment. Currently, the focus is on improving the design and ensuring style consistency. As I continue to work on this, I will refactor the files, so please ignore the code duplication for this PR.

You can compare the differences between the "Before" and "After" videos to see the improvements. The app now looks much more polished and ready to use.

I will add different designs and flexibility to configure designs in future PRs.

Please review this PR. Once it's merged, I will start working on other design enhancements. A minimalist RC-like design with multiple themes is now ready.

Thanks!

**Comment by Spiral-Memory** [comment]
Hey @sidmohanty11 

This PR is now ready for review! It mainly focuses on resolving all major UI bugs in the app and making the app consistent with theming colors, z-index, shadows, etc., as defined in the theme files. Previously, these were applied randomly, resulting in a very messy project with inconsistent colors throughout.

I've done my best to ensure everything matches across multiple themes (refer to the video). Dark mode is also supported in all themes.

You can compare the differences between the "Before" and "After" videos to see the improvements. The app now looks much more polished and ready to use.

I will add different designs and flexibility to configure designs in future PRs.

Please review this PR. Once it's merged, I will start working on other design enhancements. A minimalist RC-like design with multiple themes is now ready.

Thanks!

**Comment by Spiral-Memory** [comment]
Hey sidmohanty11

This PR is now ready for review! It mainly focuses on resolving all major UI bugs in the app and making the app consistent with theming colors, z-index, shadows, etc., as defined in the theme files. Previously, these were applied randomly, resulting in a very messy project with inconsistent colors throughout.

I've done my best to ensure everything matches across multiple themes (refer to the video). Dark mode is also supported in all themes.

You can compare the differences between the "Before" and "After" videos to see the improvements. The app now looks much more polished and ready to use.

I will add different designs and flexibility to configure designs in future PRs.

Please review this PR. Once it's merged, I will start working on other design enhancements. A minimalist RC-like design with multiple themes is now ready.

Thanks!

**Comment by Spiral-Memory** [comment]
Hey sidmohanty11

This PR is now ready for review! It mainly focuses on resolving all major UI bugs in the app and making the app consistent with theming colors, z-index, shadows, etc., as defined in the theme files. Previously, these were applied randomly, resulting in a very messy project with inconsistent colors throughout.

I've done my best to ensure everything matches across multiple themes (refer to the video). Dark mode is also supported in all themes.

You can compare the differences between the "Before" and "After" videos to see the improvements. The app now looks much more polished and ready to use.

I will add different designs and flexibility to configure designs in future PRs.

Please review this PR. Once it's merged, I will start working on other design enhancements. A minimalist RC-like design with multiple themes is now ready.

Thanks!

**Comment by Spiral-Memory** [comment]
Hey sidmohanty11

This PR is now ready for review! It mainly focuses on resolving all major UI bugs in the app and making the app consistent with theming colors, z-index, shadows, etc., as defined in the theme files. Previously, these were applied randomly, resulting in a very messy project with inconsistent colors throughout.

I've done my best to ensure everything matches across multiple themes (refer to the video). Dark mode is also supported in all themes.

You can compare the differences between the "Before" and "After" videos to see the improvements. The app now looks much more polished and ready to use.

I will add different designs and flexibility to configure designs in future PRs.

Please review this PR. Once it's merged, I will start working on other design enhancements. A minimalist RC-like design with multiple themes is now ready.

Thanks!

File: packages/react/package.json [hunk 1]
```diff
@@ -107,7 +106,6 @@
     "normalize.css": "^8.0.1",
     "prop-types": "^15.8.1",
     "react-virtuoso": "^4.3.10",
-    "reactjs-popup": "^2.0.5",
```
Comment (sidmohanty11) [inline packages/react/package.json:110]: wasn't this used for the emoji picker?

File: packages/react/src/components/Button/Button.js [hunk 2]
```diff
@@ -1,8 +1,7 @@
 import React from 'react';
-import { useTheme } from '@emotion/react';
 import PropTypes from 'prop-types';
 import useComponentOverrides from '../../theme/useComponentOverrides';
-import styles from './Button.styles';
+import useButtonStyles from './Button.styles';
 
 const getSquareSize = (size) => {
```
Comment (sidmohanty11) [inline packages/react/src/components/Button/Button.js:6]: can we move this to `Button.styles.js` so we don't need to pass it here in `Button`?

File: packages/react/src/components/Divider/Divider.styles.js [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
     height: 2px;
     margin: 0 8px 8px;
     border: 0;
     border-radius: 2px;
-    background-color: #e4e7ea;
-  `,
+    background-color: ${colors.secondary};
+  `;
+
+  return { divider };
```
Comment (sidmohanty11) [inline packages/react/src/components/Divider/Divider.styles.js:14]: is it possible to send the styles directly without destructuring them all the time?

File: packages/react/src/components/CheckBox/CheckBox.styles.js [hunk 4]
```diff
@@ -1,19 +1,26 @@
...
     cursor: pointer;
+    outline: none;
     &:active {
-      outline: 2px solid #6c727a33;
+      outline: 0.3px solid ${colors.ring};
     }
-  `,
+  `;
+
+  return { main };
```
Comment (sidmohanty11) [inline packages/react/src/components/CheckBox/CheckBox.styles.js:23]: > is it possible to send the styles directly without destructuring them all the time?

same question here

File: packages/react/src/components/DropBox/DropBox.js [hunk 5]
```diff
@@ -1,21 +0,0 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import { dropBoxStyles as styles } from './DropBox.styles';
-import { Box } from '../Box';
-
-const DropBox = ({ fullScreen = false, height, children }) => (
```
Comment (sidmohanty11) [inline packages/react/src/components/DropBox/DropBox.js:6]: this was never used?

Comment (Spiral-Memory) [reply packages/react/src/components/DropBox/DropBox.js:6]: Dropbox had a lot of bugs, causing issues while dragging and dropping. So I have removed it for the time being. The drag and drop hooks already handle it well. For the UI of drag and drop, maybe we'll introduce a new component later. This removal was necessary to ensure the project remains bug-free. So yep, currently it is not being used anywhere.

File: packages/react/package.json (hunk 1 above)
Comment (Spiral-Memory) [reply packages/react/package.json:110]: I have used normal Box for that and positioned it correctly.. So no need of this library now ! You can check in the video.. 

Also earlier, it was moving out of the boundary of EC, and was overflowing.. No such issues now !

File: packages/react/src/components/Divider/Divider.styles.js (hunk 3 above)
Comment (Spiral-Memory) [reply packages/react/src/components/Divider/Divider.styles.js:14]: Since we have to send multiple styles, it's better to send them as objects and access them easily. If we don't do that, we won't be able to send multiple styles from one hook. Even for the same component, we might have to create separate hooks for each style, which I don't think will be good at all.

File: packages/react/src/store/userStore.js [hunk 6]
```diff
@@ -38,6 +38,8 @@ const useUserStore = create((set) => ({
     set(() => ({ showCurrentUserInfo })),
   currentUser: {},
   setCurrentUser: (currentUser) => set({ currentUser }),
+  dark: false,
```
Comment (sidmohanty11) [inline packages/react/src/store/userStore.js:41]: why is this required?

Comment (Spiral-Memory) [reply packages/react/src/store/userStore.js:41]: Not required here ! 
Sorry i forgot to remove it from userStore. I already created this in theme store

File: packages/react/src/views/ChatLayout/ChatLayout.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
+import React from 'react';
```
Comment (sidmohanty11) [inline packages/react/src/views/ChatLayout/ChatLayout.js:1]: great job!

File: packages/react/src/store/userStore.js (hunk 6 above)
Comment (Spiral-Memory) [reply packages/react/src/store/userStore.js]: Removed it @sidmohanty11

File: packages/react/src/components/Button/Button.js (hunk 2 above)
Comment (Spiral-Memory) [reply packages/react/src/components/Button/Button.js:6]: Sorry Sid, didn't get your question

File: packages/react/src/views/ChatLayout/ChatLayout.js (hunk 7 above)
Comment (Spiral-Memory) [reply packages/react/src/views/ChatLayout/ChatLayout.js:1]: Thanks !

File: packages/react/src/components/Button/Button.js (hunk 2 above)
Comment (Spiral-Memory) [reply packages/react/src/components/Button/Button.js:6]: Are you talking about useButtonStyles to move somewhere else ? 
If yes, then that's not possible.. we are exporting our styles from Button.styles.js, but we must have a way to apply it to the elements.. That's why this import is necessary and then use it for the required element

File: packages/react/package.json (hunk 1 above)
Comment (sidmohanty11) [reply packages/react/package.json:110]: cool

File: packages/react/src/components/Button/Button.js (hunk 2 above)
Comment (sidmohanty11) [reply packages/react/src/components/Button/Button.js:6]: we can move this `getSquareSize` function from here to `Button.styles` and use it directly there rather than passing it from Button component

File: packages/react/src/components/DropBox/DropBox.js (hunk 5 above)
Comment (sidmohanty11) [reply packages/react/src/components/DropBox/DropBox.js:6]: got it

File: packages/react/src/components/Button/Button.js (hunk 2 above)
Comment (Spiral-Memory) [reply packages/react/src/components/Button/Button.js:6]: Ohh you are talking about getSquareSize, let me see, i think it can be done.. just a sec

Comment (Spiral-Memory) [reply packages/react/src/components/Button/Button.js:6]: Done @sidmohanty11

**Review by sidmohanty11 — COMMENTED** [review]
We might need to rethink how we structure styles. While the hooks approach is good, it may not be very feasible in the future. This looks odd that we are creating hooks to get just "styles" for each component

**Comment by Spiral-Memory** [comment]
> We might need to rethink how we structure styles. While the hooks approach is good, it may not be very feasible in the future. This looks odd that we are creating hooks to get just "styles" for each component

I also felt that it is a little odd, but I have researched it, and it's not a problem. We can't do that without hooks because we have to call useTheme(), which is a hook, and hooks can only be called inside a hook or a functional component.

There is one more way: to call useTheme() in a functional component and pass it to styles.js. However, that approach is a little messy and not clean. @sidmohanty11

**Review by sidmohanty11 — APPROVED** [review]
LGTM! Let's keep this structure for now and if we find some better approach in the future we can refactor this
